packages/transport-tcp/src/tcp.ts

```diff
@@ -121,6 +121,10 @@ export class TCP implements Transport<TCPDialEvents> {
         ...options
       }) as (IpcSocketConnectOpts & TcpSocketConnectOpts)
 
+      // Set TCP_NODELAY to true to disable Nagle's algorithm
+      // This reduces latency by sending data immediately without buffering
+      cOpts.noDelay = true
```

Comment:
`noDelay` is already set [here](https://github.com/libp2p/js-libp2p/blob/main/packages/transport-tcp/src/tcp.ts#L78)